devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,187 @@
+
+  **SOLUTION 2**: Incorrect custom path in test.firefox.path. Make sure that the path is set correctly following the example from test.template.properties
+
+4. **ERROR**: After deploying on the staging server, you get an error related to … Unsupported major.minor version ... when trying to access pages of the staged app.
+This can happen if you have Java 8 is installed and Eclipse uses Java 8 during deployment, even if you have already configured the project to use Java 7.
+Refer [this page]() to learn how to modify eclipse.ini to use Java 7 by default.
+
+5. **ERROR**: A handful of failed test cases (<10).
+
+  **SOLUTION**: Ensure compatible version of Firefox is installed as specified under Prerequisites.
+6. **ERROR**: Browser alert about invalid ssl certificate.
+
+  **SOLUTION**: Ensure that test.app.url in your test.properties uses -dot- instead of '.' when using secondary subdomains.
+  Eg. http\://4-30-dot-teammates-john.appspot.com
+7. **ERROR**: Test cases failed due to accented characters.
+
+  **SOLUTION**: Ensure that the text file encoding for your Eclipse workspace has been set to UTF-8 as specified under Prerequisites.
+8. **ERROR**: Error message in the console about ‘incorrect date format’.
+
+  **SOLUTION**: Ensure the date format of your computer matches the below. For Windows, [this link]() may be useful.
+
+  ![troubleshooting-test-5.png](images/troubleshooting-test-5.png)
+
+9. **ERROR**: InstructorCourseEnrollPageUiTest fails due to a hash mismatch.
+
+  **REASON**: The hash of the file in the repo is different from the one used in the live site due to line endings are encoded on different OSes.
+
+  **SOLUTION**: Replace “\src\main\webapp\files\Course Enroll Sample Spreadsheet.csv” with this copy of file.
+
+10. **ERROR**: "no chromedrive_osx" when testing on Mac using Chrome.
+
+  **SOLUTION**: download the file from [here]( http://chromedriver.storage.googleapis.com/index.html) and extract to /src/test/resources/lib/selenium, then rename it by adding "`_osx`". Refresh the folder in eclipse.
+
+11. **ERROR**:java.lang.UnsupportedClassVersionError when running on staging server.
+
+  **SOLUTION**: This can happen if you have Java 8 installed on your computer. Change Eclipse configuration to use Java 7 instead. Here is a useful [reference]( http://java.wildstartech.com/Java-Platform-Standard-Edition/mac-os-x-java-development/how-to-configure-eclipse-to-run-with-java-7-when-java-8-is-installed).
+12. **ERROR**: InstructorCourseDetailsPageUiTest fails in production server due to Gmail’s access restriction (e.g javax.mail.AuthenticationFailedException)
+  **SOLUTION**: “Allow access for less secure apps” for the accounts used in that particular test. As doing so will leave the accounts more vulnerable to security issues, it is strongly recommended that the access is revoked after the test passes. Check for [reference]( https://support.google.com/accounts/answer/6010255?hl=en).
+
+    # Errors specific to Linux
```

Comment:
This is not translated correctly to heading. Don't indent it.
Also, use 3 or 4 `#`s instead of just one because this is a subsection of the previous section  on "common errors and solutions".